backend/app/models.py

```diff
@@ -1,6 +1,39 @@
 """
-Models for the ***** app.
+Models for the gender analysis web app.
 """
-# from django.db import models
+from django.db import models
+from .fields import LowercaseCharField
 
-# no models yet -- write me!
+
+class Pronoun(models.Model):
+    """
+    A model that allows users to define an individual pronoun and its type
+    (e.g. subject, object, reflexive, etc). Pronouns are case-insensitive and will be
+    converted to lowercase.
+    """
+    PRONOUN_TYPES = [
+        ('subj', 'Subject'),
+        ('obj', 'Object'),
+        ('pos_det', 'Possessive determiner'),
+        ('pos_pro', 'Possessive pronoun'),
+        ('reflex', 'Reflexive'),
+    ]
+
+    pronoun = LowercaseCharField(max_length=40)
+    pronoun_type = models.CharField(max_length=7, choices=PRONOUN_TYPES)
+
+    def __repr__(self):
+        return f'Pronoun({self.pronoun, self.pronoun_type})'
+
+    def __str__(self):
+        pronoun_type_str = ''
+
+        for PRONOUN_TYPE in Pronoun.PRONOUN_TYPES:
```

Comment:
I'm not sure if this is a better way to do this but you can also do (if it is guaranteed that `self.pronoun_type`  exists in `Pronoun.PRONOUN_TYPES`):
`pronoun_types = dict(Pronoun.PRONOUN_TYPES)`
`pronoun_type_str = pronoun_types[self.pronoun_type]`

Reply:
That's a good suggestion, Shara, though I think Django expects a list of tuples in that `choices` kwarg (per [link](https://docs.djangoproject.com/en/3.2/ref/models/fields/#choices)). Following up on focusing on these lines, though, I wonder if we could use Django's built in `.get_FOO_display()` method here ([docs](https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.get_FOO_display)).

Reply:
> That's a good suggestion, Shara, though I think Django expects a list of tuples in that `choices` kwarg (per [link](https://docs.djangoproject.com/en/3.2/ref/models/fields/#choices)). Following up on focusing on these lines, though, I wonder if we could use Django's built in `.get_FOO_display()` method here ([docs](https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.get_FOO_display)).

Yes, that works well! Thanks for the tip.